core/embed/rust/src/ui/model_tt/screens.rs

```diff
@@ -8,17 +8,40 @@ use crate::ui::{
     },
 };
 
+#[cfg(feature = "new_rendering")]
+use crate::ui::{display::Color, shape::render_on_display};
+
 pub fn screen_fatal_error(title: &str, msg: &str, footer: &str) {
     let mut frame = ErrorScreen::new(title.into(), msg.into(), footer.into());
     frame.place(constant::screen());
+
+    #[cfg(feature = "new_rendering")]
+    render_on_display(None, Some(Color::black()), |target| {
+        frame.render(target);
+    });
+
+    #[cfg(not(feature = "new_rendering"))]
     frame.paint();
     display::refresh();
 }
 
 pub fn screen_boot_stage_2() {
     let mut frame = WelcomeScreen::new(false);
     frame.place(screen());
-    display::sync();
-    frame.paint();
-    display::refresh();
+
+    #[cfg(feature = "new_rendering")]
+    {
+        display::sync();
+        render_on_display(None, Some(Color::black()), |target| {
+            frame.render(target);
+        });
+        display::refresh();
+    }
+
+    #[cfg(not(feature = "new_rendering"))]
+    {
+        display::sync();
+        frame.paint();
+        display::refresh();
+    }
 }
```

Comment:
yes, they are with the small exception that `screen_fatal_error()` does not sync with the display, but IMHO it could/should. So I simplified it. see d57fc99bd4c2317d01b654e4ac77332ee20ecd38